src/L1/rollup/ScrollChain.sol

```diff
@@ -157,6 +220,31 @@ contract ScrollChain is OwnableUpgradeable, PausableUpgradeable, IScrollChain {
     /// @inheritdoc IScrollChain
     mapping(uint256 => bytes32) public override withdrawRoots;
 
+    /// @notice Mapping from batch index to batch committed timestamp.
+    /// @dev This is enabled after Euclid upgrade.
+    mapping(uint256 => uint256) public batchCommittedTimestamp;
+
+    /// @inheritdoc IScrollChain
+    /// @dev This is enabled after Euclid upgrade.
+    uint256 public override lastTeeVerifiedBatchIndex;
+
+    /// @notice The mask for enabled proof types.
+    /// @dev This is enabled after Euclid upgrade.
+    uint256 public enabledProofTypeMask;
+
+    /// @notice The state for mismatched batch.
+    /// @dev This is enabled after Euclid upgrade.
+    UnresolvedState public unresolvedState;
+
+    /// @notice The state for bundle size.
+    /// @dev This is enabled after Euclid upgrade.
+    /// @dev Assume the list is `[(s[1], b[1]), (s[2], b[2]), ..., (s[n], b[n])]`,  where `s[i]` is the bundle size
+    ///      and `b[i]` is the start batch index. Then for each `i > 1`, we should have `b[i] > b[i - 1]` and
+    ///      `(b[i] - b[i - 1]) % s[i - 1] = 0`.
+    ///      If a bundle has end batch range `[x, y]`, we need to find last `i` such that `y > b[i]`. And the following
+    ///      should be satisfied: `(y - b[i]) % s[i] = 0` and `y - x + 1 = s[i]`.
+    BundleSizeStruct[] public bundleSize;
```

Comment:
does this mean rollup-relayer tee-prover would read the slots to decide bundle size? since currently tee-prover is not only an event-listening client.

Reply:
for rollup-relayer, local configuration is fine, since only us run it. Tee host can read from contract.